posix/unix.c

```diff
@@ -26,11 +26,11 @@
 #define US_MIN_BUFFER_SIZE SIZE_PAGE
 #define US_MAX_BUFFER_SIZE 65536
 
-#define US_BOUND      (1 << 0)
-#define US_LISTENING  (1 << 1)
-#define US_ACCEPTING  (1 << 2)
-#define US_CONNECTING (1 << 3)
-
+#define US_BOUND        (1 << 0)
+#define US_LISTENING    (1 << 1)
+#define US_ACCEPTING    (1 << 2)
+#define US_CONNECTING   (1 << 3)
+#define US_DISCONNECTED (1 << 4) /* NOTE: invalidates connect pointer */
```

Comment:
Hm... new socket is not in `disconnected` state, it's a special flag that indicates the socket was already connected and than disconnected from the other side (sooooo.... it's "disconnecting" now as "disconnected" sockets have state == 0?)

Reply:
It should really be US_PEER_DISCONNECTED but I've decided it's too long.